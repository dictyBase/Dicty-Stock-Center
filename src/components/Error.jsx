import React, { Component } from 'react'
import 'styles/core.scss'

export default class Error extends Component {
    displayName = 'homepage component';
    render() {
        const { auth, order } = this.props
        const orderError = order.error
        const authError = auth.error
        return (
            <div className="container">
                <div className="alert alert-danger text-center">
                    <h1><i className="fa fa-exclamation-triangle" ></i> Error</h1>
                    <h2>{ authError && authError }</h2>
                    <h2>{ orderError && orderError }</h2>
                </div>
            </div>
        )
    }
}
