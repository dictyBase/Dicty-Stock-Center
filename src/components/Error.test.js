import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { BrowserRouter } from "react-router-dom"
import { Error } from "./Error"

test("matching a snapshot of Error", () => {
  const auth = {
    error: "404"
  }
  const order = {
    error: "404"
  }
  const page = {
    error: "404"
  }

  const component = renderer.create(
    <BrowserRouter>
      <Error auth={auth} order={order} page={page} />
    </BrowserRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
