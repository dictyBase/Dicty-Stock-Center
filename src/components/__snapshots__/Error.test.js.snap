// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Error 1`] = `
.c0 {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

.c1 {
  padding: 15px;
  margin-bottom: 20px;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #a94442;
  background-color: #f2dede;
  border-color: #ebccd1;
  text-align: center;
}

@media (min-width:768px) {
  .c0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .c0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 1170px;
  }
}

<div
  className="c0"
>
  <div
    className="sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="csVOHI"
      is={null}
    >
      <div
        className="c1"
      >
        <h1>
          <span
            aria-hidden={true}
            className="fa fa-exclamation-triangle"
          />
           Error
           
          <span
            aria-hidden={true}
            className="fa fa-exclamation-triangle"
          />
        </h1>
        <h3 />
        <h3 />
        <h3 />
        <p>
          Please 
          <a
            href="/contact"
            onClick={[Function]}
          >
            contact us
          </a>
           if this problem persists.
        </p>
        <p>
          <a
            href="/"
            onClick={[Function]}
          >
            Return Home
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
